packages/suite/src/views/recovery/index.tsx

```diff
@@ -87,116 +83,55 @@ export const Recovery = ({ onCancel }: ForegroundAppProps) => {
                 );
-            }
             case 'select-word-count':
-                return (
-                    <>
-                        <H3 margin={{ bottom: spacings.md }}>
-                            <Translation id="TR_SELECT_NUMBER_OF_WORDS" />
-                        </H3>
-                        <SelectWordCount onSelect={onSetWordsCount} />
-                    </>
-                );
+                return <SelectWordCountStep setWordCount={setWordCount} wordCount={wordCount} />;
             case 'select-recovery-type':
                 return (
-                    <>
-                        <H3 margin={{ bottom: spacings.md }}>
-                            <Translation id="TR_CHOOSE_RECOVERY_TYPE" />
-                        </H3>
-                        <SelectRecoveryType onSelect={onSetRecoveryType} />
-                    </>
+                    <SelectRecoveryTypeStep
+                        setRecoveryType={setRecoveryType}
+                        recoveryType={recoveryType}
+                    />
                 );
-            case 'in-progress':
             case 'waiting-for-confirmation':
+            case 'in-progress':
+                // When T1B1 requests PIN confirmation, the status is still 'waiting-for-confirmation'
+                // so we need to check the modal context to know if we should show the loading indicator
                 if (modal.context !== MODAL.CONTEXT_DEVICE) {
-                    return <Loading />;
+                    return (
+                        <Box padding={spacings.xxl}>
+                            <Loading />
+                        </Box>
+                    );
                 }
 
                 // Do not rely on Capability_PassphraseEntry feature. For ancient firmwares it's not there,
                 // and we want to allow devices that have unsupported FW to be able to check the seed.
-                if (device.features.internal_model === DeviceModelInternal.T1B1) {
-                    return <T1B1InputStep />;
+                if (isT1B1) {
+                    switch (modal.windowType) {
+                        case UI.REQUEST_PIN:
+                            return (
+                                <PinMatrix pin={pin} setPin={setPin} onSubmit={handlePinSubmit} />
+                            );
+                        case 'ButtonRequest_Other':
+                            return <RequestConfirmationStep />;
+                        case 'WordRequestType_Plain':
+                            return <WordInputStep />;
+                        case 'WordRequestType_Matrix6':
+                            return <WordInputAdvanced count={6} />;
+                        case 'WordRequestType_Matrix9':
+                            return <WordInputAdvanced count={9} />;
+                        // to satisfy TS, should not happen
+                        default:
+                            return null;
```

Comment:
TS passes without it :heavy_check_mark: which is good because these defaults are not nice, can lead to unexpected bugs in future.
```suggestion
                        // to satisfy TS, should not happen
                        default:
                            return null;
```